AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -502,23 +511,190 @@ - (void)testPerformCosmosDbAsyncOperationWithHttpClient {
         completionHandler(actualData, nil, nil);
       });
   NSDictionary *dic = @{@"abv" : @1, @"foo" : @"bar"};
+  MSMockDocument *mockDoc = [[MSMockDocument alloc] initFromDictionary:dic];
   __block NSData *data = [NSJSONSerialization dataWithJSONObject:dic options:0 error:nil];
 
   // When
   [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:httpClient
                                               tokenResult:tokenResult
                                                documentId:kMSDocumentIdTest
                                                httpMethod:kMSHttpMethodGet
-                                                     body:data
+                                                 document:mockDoc
                                         additionalHeaders:nil
+                                        additionalUrlPath:kMSDocumentIdTest
                                         completionHandler:handler];
 
+  NSError *error;
+  NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:(NSData *)actualData options:0 error:&error];
+  NSDictionary *document = dictionary[@"document"];
+  NSData *actualDocumentAsData = [NSJSONSerialization dataWithJSONObject:document options:0 error:nil];
+
   // Then
   XCTAssertTrue(completionHandlerCalled);
-  XCTAssertEqualObjects(data, actualData);
+  XCTAssertEqualObjects(data, actualDocumentAsData);
   XCTAssertEqualObjects(expectedURLString, [actualURL absoluteString]);
 }
 
+- (void)testPerformCosmosDbAsyncOperationWithNilDocument {
+
+  // If
+  MSHttpClient *httpClient = OCMClassMock([MSHttpClient class]);
```

Comment:
I'm really curious about this one.
It requires `[httpClient stopMocking]` at the end of each test if it is class-mocked but I don't see any of those in the file except two unit tests. This could produce unexpected side effect since `MSHttpClient` is class mocked, it can be instantiated within other unit tests. Please at least `stopMocking` all in the unit tests that you wrote here.

Reply:
the http client it created in the local scope and is not set int the `sut`, so  we don't need to stop it